### PR TITLE
OTG Test - TE-11.2: Backup NHG: Multiple NH

### LINKED ISSUE #646: OTG: TE-11.2: Backup NHG: Multiple NH

### PR DESCRIPTION
**Pull request by octpetre**
OTG conversion of **_TE-11.2: Backup NHG: Multiple NH_**
Output log found [here](https://drive.google.com/file/d/1RwoXXSC0UCy9Uqc8JsPmQSfa_nW-e9i3/view?usp=share_link)
Fixes #646

### CODE REVIEW

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go [hunk 1]
```diff
@@ -264,67 +247,73 @@ func testIPv4BackUpSwitch(ctx context.Context, t *testing.T, args *testArgs) {
...
-		WithDstEndpoints(dstEndPoint...).
-		WithHeaders(ondatra.NewEthernetHeader(), hdr).WithFrameSize(300)
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config, name string) string {
+
+	flow := top.Flows().Add().SetName(name)
+	flow.Metrics().SetEnable(true)
+	e1 := flow.Packet().Add().Ethernet()
+	e1.Src().SetValue(atePort1.MAC)
+	flow.TxRx().Port().SetTxName(atePort1.Name)
+	waitOTGARPEntry(t)
```
Comment (jasdeep-hundal) [inline feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:277]: Move this to just happen once after starting protocols in the TestBackup function?

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go [hunk 2]
```diff
@@ -264,67 +247,73 @@ func testIPv4BackUpSwitch(ctx context.Context, t *testing.T, args *testArgs) {
...
+	dstMac := ate.OTG().Telemetry().Interface(atePort1.Name + ".Eth").Ipv4Neighbor(dutPort1.IPv4).LinkLayerAddress().Get(t)
+	e1.Dst().SetChoice("value").SetValue(dstMac)
+	v4 := flow.Packet().Add().Ipv4()
+	v4.Src().SetValue(atePort1.IPv4)
+	v4.Dst().Increment().SetStart(dstPfxMin).SetCount(1)
```
Comment (jasdeep-hundal) [inline feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:282]: I'm not sure why we set a count in the equivalent ATE test, can we just use 'SetValue' here?

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go [hunk 3]
```diff
@@ -264,67 +247,73 @@ func testIPv4BackUpSwitch(ctx context.Context, t *testing.T, args *testArgs) {
...
+	ate.OTG().PushConfig(t, top)
+	ate.OTG().StartProtocols(t)
```
Comment (jasdeep-hundal) [inline feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:284]: Is this needed?

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go [hunk 4]
```diff
@@ -264,67 +247,73 @@ func testIPv4BackUpSwitch(ctx context.Context, t *testing.T, args *testArgs) {
...
-		WithSrcEndpoints(srcEndPoint).
-		WithDstEndpoints(dstEndPoint...).
-		WithHeaders(ondatra.NewEthernetHeader(), hdr).WithFrameSize(300)
+func createFlow(t *testing.T, ate *ondatra.ATEDevice, top gosnappi.Config, name string) string {
+
+	flow := top.Flows().Add().SetName(name)
+	flow.Metrics().SetEnable(true)
+	e1 := flow.Packet().Add().Ethernet()
+	e1.Src().SetValue(atePort1.MAC)
+	flow.TxRx().Port().SetTxName(atePort1.Name)
```
Comment (jasdeep-hundal) [inline feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:276]: What the behavior if there is no Rx name set here?

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go [hunk 5]
```diff
@@ -264,67 +247,73 @@ func testIPv4BackUpSwitch(ctx context.Context, t *testing.T, args *testArgs) {
...
 		}
 	}
 }
 
 // flapinterface shut/unshut interface, action true bringsup the interface and false brings it down
-func flapinterface(t *testing.T, ate *ondatra.ATEDevice, port string, action bool) {
-	ateP := ate.Port(t, port)
-	ate.Actions().NewSetPortState().WithPort(ateP).WithEnabled(action).Send(t)
+func flapinterface(t *testing.T, dut *ondatra.DUTDevice, port string, action bool) {
+	// When running on kne the corresponding DUT port will be taken down
```
Comment (jasdeep-hundal) [inline feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:311]: I would rephrase this/also mention that trying to shut OTG ports in KNE has no effect.

**Review by jasdeep-hundal — CHANGES_REQUESTED** [review]

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go [hunk 6]
```diff
@@ -359,3 +348,12 @@ func aftCheck(t testing.TB, dut *ondatra.DUTDevice, prefix string, expectedNH []
 		}
 	}
 }
+
+// Waits for at least one ARP entry on the tx OTG interface
+func waitOTGARPEntry(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (octpetre) [reply feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:353]: Done

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go (hunk 3 above)
Comment (octpetre) [reply feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:284]: After every push config we need to include the Start Protocols in order to have this test working fine on hw

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go (hunk 2 above)
Comment (octpetre) [reply feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:282]: Replaced with SetValue

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go (hunk 1 above)
Comment (octpetre) [reply feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:277]: Moved in testIPv4BackUpSwitch function

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go (hunk 4 above)
Comment (octpetre) [reply feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:276]: In this case flow statistics could not be retrieved on the OTG side but the traffic will be generated. We configure port 1 as Tx and the packets may be received by one or more Rx Ports. At this time OTG cannot accept more than 1 destination as Rx.

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go (hunk 5 above)
Comment (octpetre) [reply feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:311]: Done

File: feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go (hunk 3 above)
Comment (jasdeep-hundal) [reply feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:284]: Couple questions:
- Is that specific to hardware? (If so, worth a comment.)
- What specifically happens with protocol state in this case? (Eg. it seems we're not restarting everything since we're not waiting for the ARP entry again)

Comment (octpetre) [reply feature/gribi/otg_tests/backup_nhg_multiple_nh_test/backup_nhg_multiple_nh_test.go:284]: After every PushConfig, StartProtocols is required to run the test on hardware. There without the protocols (control plane) started, we can't apply/start the traffic items where ARP has to be resolved.

**Review by jasdeep-hundal — APPROVED** [review]
